libraries/chain/db_block.cpp

```diff
@@ -377,17 +366,40 @@ signed_block database::_generate_block(
    // the value of the "when" variable is known, which means we need to
    // re-apply pending transactions in this method.
    //
+
+   // pop pending state (reset to head block state)
    _pending_tx_session.reset();
+
+   // Check witness signing key
+   if( !(skip & skip_witness_signature) )
+   {
+      // Note: if this check failed (which won't happen in normal situations),
+      // we would have temporarily broken the invariant that
+      // _pending_tx_session is the result of applying _pending_tx.
```

Comment:
At this point _pending_tx_session has been cleared, so the comment makes no sense.

Reply:
Yes, `_pending_tx_session` has been cleared, but `_pending_tx` hasn't, so they're inconsistent.